packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -921,6 +938,30 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
         _crawlingContext: Context,
     ) {}
 
+    protected _handleRequestWithDelay(request:Request, source: RequestQueue | RequestList)
+    {
+        const domain = getDomain(request.url);
+        if (domain) {
```

Comment:
can this be really falsy value? even if so, the if statement should check the opposite instead of nesting the whole code